README.md

```diff
@@ -16,17 +16,22 @@ Once you have implemented the `sa` program (and tested it to the best of your ab
 # Report
 
 ## Algorithm
-
+Started with SAIS, abandoned it as i didn't get the slides (and no good youtube videos exist for the recursive part). Did the naive instead. 
+Did the match algorithm using an LCP array, constructed naively.
 *Which algorithm did you use to construct the suffix array. What is its running time?*
 
 ## Insights you may have had while implementing the algorithm
 
+
 ## Problems encountered if any
+Wasn't able to implement SAIS due to lack of understanding of the recursive part (it did go well up to that point though).
 
 ## Validation
-
+Ran on several types of strings and patterns, including trusty missisippi.
 *How did you validate that the construct and the search algorithm works?*
 
 ## Running time
-
+![](match_dna.png)
+![](SA_DNA.png)
+As can be seen on the figures the SA construction rises exponentially as length increases and the match is slowest when the pattern is short, meaning there should be more matches in a random, non-repeating string. Unfortunately, as the LCP construction algorithm is veery slow, i wasn't able to get quality results for strings and patterns of smaller alphabets.
```

Comment:
If the time grows exponential there is something seriously wrong, as it is almost trivial to get a quadratic time algorithm to work.

I'll have a look at the LCP algorithm, but that is also trivial to compute in $O(n^2)$ so it shouldn't take that long either. (It is a little trickier to get it in $O(n)$ but it is not a major challenge compared to something like the suffix trees).

If the running time is highest for smaller strings, that should be because there is more output, but the output should never take longer than $O(z)$ so I don't know what is happening there either.